File: backend/app/schemas/alert.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict

__all__ = ["AlertCreate", "AlertUpdate", "AlertResponse"]


class AlertCreate(BaseModel):
    employee_id: str
    prediction_id: int | None = None
    severity: str = "medium"
    title: str
    description: str | None = None
    notes: str | None = None


class AlertUpdate(BaseModel):
    status: str | None = None
    assigned_to: str | None = None
    notes: str | None = None
    severity: str | None = None


class AlertResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    prediction_id: int | None = None
    employee_id: str
    severity: str
    status: str
    assigned_to: str | None = None
    title: str
    description: str | None = None
    notes: str | None = None
    created_at: datetime
    updated_at: datetime | None = None
    resolved_at: datetime | None = None

File: frontend/src/types/alert.ts
export type AlertSeverity = "low" | "medium" | "high" | "critical";
export type AlertStatus = "open" | "investigating" | "resolved" | "false_positive";

export interface Alert {
  id: number;
  prediction_id?: number | null;
  employee_id: string;
  severity: AlertSeverity | string;
  status: AlertStatus | string;
  assigned_to?: string | null;
  title: string;
  description?: string | null;
  notes?: string | null;
  created_at: string;
  updated_at?: string;
  resolved_at?: string | null;
}

export interface AlertCreate {
  employee_id: string;
  title: string;
  severity?: string;
  prediction_id?: number;
  description?: string;
  notes?: string;
}

export interface AlertUpdate {
  status?: string;
  severity?: string;
  assigned_to?: string;
  notes?: string;
}
